File: Assets/Scripts/Module/Invoke/FActions/WaitFrameAction.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace FAction
{
    public class WaitFrameAction : FAction
    {
        public override ActionType ActionType => ActionType.WaitFrame;
        // 延迟多少帧
        public int waitCount;

        int remain = -1;

        public override void Tick()
        {
            base.Tick();
            if (remain < 0)
            {
                remain = waitCount;
            }

            remain--;

            if (remain < 0)
            {
                this.IsFinish = true;
            }
        }

        public override void Clear()
        {
            base.Clear();
            waitCount = 0;
        }

        public override void Replay()
        {
            base.Replay();
            remain = -1;
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/FAction/WaitFrameAction.lua
---@class CS.FAction.WaitFrameAction : CS.FAction.FAction
CS.FAction.WaitFrameAction = {}

---@field public CS.FAction.WaitFrameAction.waitCount : CS.System.Int32
CS.FAction.WaitFrameAction.waitCount = nil

---@property readonly CS.FAction.WaitFrameAction.ActionType : CS.FAction.ActionType
CS.FAction.WaitFrameAction.ActionType = nil

---@return CS.FAction.WaitFrameAction
function CS.FAction.WaitFrameAction()
end

function CS.FAction.WaitFrameAction:Tick()
end

function CS.FAction.WaitFrameAction:Clear()
end

function CS.FAction.WaitFrameAction:Replay()
end